drafts/bitcartel-payment-disclosure/draft.rst

```diff
@@ -0,0 +1,345 @@
+
+When a transaction involves only transparent addresses, the sender and recipient can verify payment by examining the blockchain.  A third party can also perform this verification if they know the transparent addresses of the involved parties.
+
+However, if the transaction involves shielded addresses, the blockchain by itself does not contain enough information from which a record of the payment can be reconstructed and verified.
+
+Let us examine the types of transaction that might occur and when a method of payment disclosure would be useful:
+
+transparent --> transparent
+  The source, destination and amount are visible on the public blockchain.  Payment disclosure is not required.
+
+transparent --> shielded
+  The destination address and amount sent to that address cannot be confirmed. Payment disclosure is required.
+
+shielded --> transparent
+  The recipient can see the amount received at their destination address, but cannot identify the sender.  Payment disclosure is required.
+
+shielded --> shielded
+  The sender, recipient and amount are unknown.  Payment disclosure required.
+
+TODO: Perhap use a table to make this easier to understand, from perspective of sender,recipient and third party?
+
+Design Overview
+---------------
+
+Allow the sender to automatically or optionally create a payment disclosure when creating a transaction involving shielded addresses.
+
+The sender or a third party can present this payment disclosure to the recipient to prove that funds were sent to their address.
+
+Example
+-------
+A group of coworkers go out for dinner and decide to split the bill.  The restaurant only accepts physical cash.  Alice ends up paying the entire bill as her coworkers have no cash on them.
+
+The next day Bob sends 1 ZEC to Alice's shielded address for his share of the bill.  Other coworkers do the same.  A week later Alice remembers that she is owed money so she checks her shielded address.  Alice sees that the balance is less than what it should be, so she asks her coworkers to confirm their payments.
+
+Bob retrieves his payment disclosure and emails it to Alice.  Alice is travelling and only has access to her smartphone.  She visits a Zcash blockchain explorer and copies the payment disclosure into the search box. The explorer confirms that Bob paid his share.
+
+Charlie attempts to retrieve the payment disclosure but can't find one.  Charlie had used the command line to send his funds to Alice, but never checked the status of the z_sendmany operation.  Checking the debug logs he discovers that the operation failed due to an incorrect parameter.  Charlie tells Alice that he will re-send his payment.
+
+
+Design Considerations
+---------------------
+The payment disclosure does not prove that the party presenting the payment disclosure is the sender.
+
+To mitigate against a replay attack, the recipient could enforce a policy where the first person to present a given payment disclosure is the honest sender.
```

Comment:
Delete this sentence; that's not a reliable way of doing it. It could be okay if it's sent over a secure channel but it's tricky to explain the resulting security properties.

Reply:
ok